wildfly-mcp/subsystem/src/main/java/org/wildfly/extension/mcp/deployment/McpSseHandlerServiceInstaller.java

```diff
@@ -62,10 +106,72 @@ public void run() {
                 ROOT_LOGGER.endpointUnregistered(ssePath, host.get().getName());
             }
         };
-        ServiceInstaller.builder(start, stop)
-                .requires(List.of(host, ServiceDependency.on(Capabilities.MCP_SERVER_PROVIDER_DESCRIPTOR)))
-                .asActive()
-                .build()
-                .install(context);
+        if (oidcSecured) {
+            ServiceInstaller.builder(start, stop)
+                    .requires(List.of(host, ServiceDependency.on(Capabilities.MCP_SERVER_PROVIDER_DESCRIPTOR), securityDomain))
+                    .asActive()
+                    .build()
+                    .install(context);
+        } else {
+            ServiceInstaller.builder(start, stop)
+                    .requires(List.of(host, ServiceDependency.on(Capabilities.MCP_SERVER_PROVIDER_DESCRIPTOR)))
+                    .asActive()
+                    .build()
+                    .install(context);
+        }
+    }
+
+    private HttpServerAuthenticationMechanismFactory getHttpServerAuthenticationMechanismFactory(DeploymentUnit deploymentUnit) {
+        WarMetaData warMetaData = deploymentUnit.getAttachment(WarMetaData.ATTACHMENT_KEY);
+        if (warMetaData == null) {
+            return null;
+        }
+        String json = getJsonConfiguration(warMetaData);
+        if (json != null) {
```

Comment:
What does the json config look like?

Reply:
Hum, I don't use the json file, I create a json from the subsystem configuration

Reply:
/subsystem=elytron-oidc-client/secure-deployment=ROOT.war:add(client-id=mcp-client, provider-url="${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm", ssl-required=EXTERNAL, public-client="true", principal-attribute="preferred_username")

Reply:
I will add this in the layer